File: 27-RemoveElement/27-RemoveElement.cpp
// Last updated: 8/29/2025, 9:53:15 PM
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int start = 0;
        int i = 0;
        int count = 0;
        while(i < nums.size()){
            if(nums[i] == val){
                i++;
                count++;
            } else {
                nums[start] = nums[i];
                start++;
                i++;
            }
        }
        return nums.size()-count;
    }
};

File: 27-RemoveElement/27-RemoveElement.py
# Last updated: 8/29/2025, 9:53:19 PM
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        j = 0 
        count = len(nums)
        for i in range(len(nums)):
            if nums[i] == val:
                count -= 1
                continue 
            nums[j] = nums[i]
            j += 1
        return count 